pandas/core/construction.py

```diff
@@ -377,7 +377,10 @@ def extract_array(obj, extract_numpy: bool = False):
     array([1, 2, 3])
     """
     if isinstance(obj, (ABCIndexClass, ABCSeries)):
-        obj = obj.array
+        arr = obj._values
+        if not extract_numpy and isinstance(arr, np.ndarray):
+            return obj.array
```

Comment:
Yes, I was also just thinking while looking above at the `.array` implementation, that we could do the same here instead of going through the ".array -> wrap in PandasArray -> extract the numpy array again" route, that will further reduce some overhead of `extract_array(..., extract_numpy=True)`.

Could also do a `arr = PandasArray(arr)` here for being explicit (it's not that it duplicates a lot from `.array`)